src/Compilers/Core/MSBuildTask/Vbc.cs

```diff
@@ -960,6 +954,40 @@ private bool InitializeHostCompiler(IVbcHostObject vbcHostObject)
             return true;
         }
 
+        // VbcHostObject doesn't support VB versions beyond 15,
+        // so the LangVersion will be passed through ICompilerOptionsHostObject.SetCompilerOptions instead
+        private static bool DeferToICompilerOptionsHostObject(string langVersion, IVbcHostObject vbcHostObject)
+        {
+            if (!(vbcHostObject is ICompilerOptionsHostObject))
+            {
+                return false;
+            }
+
+            if (langVersion == null)
+            {
+                // CVbcMSBuildHostObject::SetLanguageVersion can handle null
+                return false;
+            }
+
+            // CVbcMSBuildHostObject::SetLanguageVersion can handle version below 15
```

Comment:
Minor point: Should this be `up to 15` since it looks like `15` is also handled.